compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1644,6 +1644,12 @@ error.invalid.inclusion.with.mismatched.qualifiers=\
 error.invalid.reference.with.mismatched.qualifiers=\
   invalid type reference: missing ''{0}'' qualifier(s) in the referencing object constructor expression
 
+error.invalid.read.only.typedesc.inclusion.in.object.typedesc=\
+  object type inclusion cannot be used with a ''readonly'' type descriptor in an ''object'' type descriptor
+
+error.invalid.read.only.typedesc.inclusion.in.non.read.only.class=\
+  object type inclusion cannot be used with a ''readonly'' type descriptor in a ''class'' that is not ''readonly''
```

Comment:
I think this message is unclear. When we say `that is not` here that means the class including the object. But we don't say we cannot include because we are trying to include a readonly object.
Isn't it clear just saying something similar to "Immutable type descriptor can only include immutable types"

Reply:
Or can we put the readonly type descriptor name here.

`object type inclusion cannot be used with a 'readonly' type descriptor in a 'class' that is not 'readonly'`

`object type inclusion cannot be used with a 'readonly' type descriptor `Timestamp` in a 'class' that is not 'readonly'`

```Ballerina
public type Timestamp readonly & object {
    int value;

    public function toMillisecondsInt() returns int;
    public function toString() returns string;
};

class TimestampImpl  {
    *Timestamp;

    function init(decimal value) {
        self.value = <int> value;
    }

    public function toMillisecondsInt() returns int => self.value;

    public function toString() returns string => "test";
}
```

Reply:
> I think this message is unclear. When we say `that is not` here that means the class including the object. But we don't say we cannot include because we are trying to include a readonly object.

It's an error specifically because it is trying to include `readonly` that is part of an intersection, which I'm trying to convey via the "object type inclusion cannot be used with a ''readonly'' type descriptor". 

> Isn't it clear just saying something similar to "Immutable type descriptor can only include immutable types"

Doesn't this imply that we cannot use inclusion with a mutable type in a class that is immutable? But that is not the case. 

> Or can we put the readonly type descriptor name here.

Since the error is specifically for `readonly` via `Timestamp` and not directly for `Timestamp`, IMO it's not quite accurate to include the name.

Reply:
Let's change the error in a separate PR, if we can come up with a better error message.